src/Corax/Constants.cs

```diff
@@ -19,7 +19,11 @@ static Constants()
             }
         }
         
-        
+        public static class Terms
+        {
+            public const int MaxLength = 512;
```

Comment:
Why `512`? Max key is `1024`
https://github.com/ravendb/ravendb/blob/5c248d7f3f9c82cacc53e2570baf7e741caa0280/src/Voron/Constants.cs#L65-L75

Reply:
To make sure that we have more space, basically.
Practically speaking, there isn't much need for that big a term

Reply:
This is maximum key for the `CompactTree` itself.